### ISSUE TITLE
Upon receiving the second Great Admiral, the AI activates both

### ISSUE THREAD

**Issue by nobubblegums** [LoneGazebo/Community-Patch-DLL#8413]
_1. Mod version (X.Y.Z, e.g. 1.2.1):_
1.2.1

_2. Mod list (if using Vox Populi only, leave blank):_
Vox populi + Test.sql (to make early buildings give admirals for easier checking)

[Test.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/7605817/Test.zip)
_3. Error description:_ 
When in peace, upon receiving the second Great Admiral, the AI activates both at the same time, even if it has 100% happiness.

_4. Steps to reproduce (optional):_

---------------------------
Supporting information:
Please note that you can attach .zip files by dragging-and-dropping them. If possible, zip up all supporting data and post that way.

1. Log files (always attach your Logs folder, located at My Documents/My Games/Sid Meier's Civilization 5. For instructions, go to the repository's main page, under "To enable logging for bug reports"):

[Logs.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/7605909/Logs.zip)

2. Save game (always attach a save that was made a turn before the error; located at My Documents/My Games/Sid Meier's Civilization 5/ModdedSaves; you can change autosave frequency in the game's Options menu):
[AutoSave_0001 BC-3960.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/7605910/AutoSave_0001.BC-3960.zip)

3. CvMiniDump.dmp file (attach if experiencing a game crash. Located at Program Files/Steam/steamapps/common/Sid Meier's Civilization V):

4. Screenshots (optional):

**Comment by nobubblegums** [comment]
I think I may have found the issue -which may affect all GPs when there are more than one of them, not just the GA. In CvPlayerAI ProcessGreatPeople, GPs get their directives in the same loop, and then their directives are set and activated. The GetDirectiveAdmiral function seem to imply that one GA should be kept in reserve as long as the empire isn't unhappy:

	int iThreshold = IsEmpireUnhappy() ? 0 : 1;
	if (iGreatAdmiralCount > iThreshold && pGreatAdmiral->canGetFreeLuxury())
	{
		return GREAT_PEOPLE_DIRECTIVE_USE_POWER;
	}

But when there are more than one Great Admirals, this function doesn't work as intended because ProcessGreatPeople first gets directives for all units and counts, then sets the directive and processes the great people, so iGreatAdmiralCount returns greater than iThreshold for all Admirals and all of them use power on that turn. 
I tried a built where I inserted 'pLoopUnit->SetGreatPeopleDirective(eDirective)' after each GetDirective under the switch statement, and that fixed the issue. I doubt that's the proper way of doing this though, so I didn't commit that.
